Sources/Datadog/Datadog.swift

```diff
@@ -92,49 +92,69 @@ public class Datadog {
+        let internalLoggerConfiguration = InternalLoggerConfiguration(
+            applicationVersion: validConfiguration.applicationVersion,
+            environment: validConfiguration.environment,
             userInfoProvider: userInfoProvider,
             networkConnectionInfoProvider: networkConnectionInfoProvider,
             carrierInfoProvider: carrierInfoProvider
         )
 
-        let tracing = TracingFeature(
-            directory: try obtainTracingFeatureDirectory(),
-            configuration: validConfiguration,
-            performance: performance,
-            loggingFeatureAdapter: LoggingForTracingAdapter(loggingFeature: logging),
-            mobileDevice: mobileDevice,
-            httpClient: httpClient,
-            tracesUploadURLProvider: UploadURLProvider(
-                urlWithClientToken: validConfiguration.tracesUploadURLWithClientToken,
-                queryItemProviders: [
-                    .batchTime(using: dateProvider)
-                ]
-            ),
-            dateProvider: dateProvider,
-            tracingUUIDGenerator: DefaultTracingUUIDGenerator(),
-            userInfoProvider: userInfoProvider,
-            networkConnectionInfoProvider: networkConnectionInfoProvider,
-            carrierInfoProvider: carrierInfoProvider
-        )
+        userLogger = createSDKUserLogger(configuration: internalLoggerConfiguration)
+        developerLogger = createSDKDeveloperLogger(configuration: internalLoggerConfiguration)
+
+        // Then, initialize features:
+
+        let httpClient = HTTPClient()
+        let mobileDevice = MobileDevice.current
+
+        var logging: LoggingFeature?
+        var tracing: TracingFeature?
+
+        if configuration.loggingEnabled {
+            logging = LoggingFeature(
+                directory: try obtainLoggingFeatureDirectory(),
+                configuration: validConfiguration,
+                performance: performance,
+                mobileDevice: mobileDevice,
+                httpClient: httpClient,
+                logsUploadURLProvider: UploadURLProvider(
+                    urlWithClientToken: validConfiguration.logsUploadURLWithClientToken,
+                    queryItemProviders: [
+                        .ddsource(),
+                        .batchTime(using: dateProvider)
+                    ]
+                ),
+                dateProvider: dateProvider,
+                userInfoProvider: userInfoProvider,
+                networkConnectionInfoProvider: networkConnectionInfoProvider,
+                carrierInfoProvider: carrierInfoProvider
+            )
+        }
+
+        if configuration.tracingEnabled {
+            tracing = TracingFeature(
+                directory: try obtainTracingFeatureDirectory(),
+                configuration: validConfiguration,
+                performance: performance,
+                loggingFeatureAdapter: logging.flatMap { LoggingForTracingAdapter(loggingFeature: $0) },
```

Comment:
do we need `flatMap` here?

Reply:
Yup, the `logging` is an optional `LoggingFeature?`, so it's either `flatMap {}` or we'd have to wrap the entire 18-lines `TracingFeature()` setup within a `if let`, which is IMO less readable.

Reply:
if you are happy with `flatMap` we can keep it, otherwise we can do this instead:
```swift
var adapter: Adapter? = nil
if let someLogging = logging {
  adapter = Adapter(someLogging)
}
...
loggingFeatureAdapter: adapter
...
```

Reply:
IMO it's fine with, we don't get `flatMap` wrong:
> Evaluates the given closure when this `Optional` instance is not `nil`, passing the unwrapped value as a parameter.

https://developer.apple.com/documentation/swift/optional/1540500-flatmap